deepchem/feat/molecule_featurizers/molgan_featurizer.py

```diff
@@ -0,0 +1,214 @@
+        bond_labels: List[RDKitBond] = None,
+        atom_labels: List[int] = None,
+    ):
+        """
+        Parameters
+        ----------
+        max_atom_count: int, default 9
+            Maximum number of atoms used for creation of adjacency matrix.
+            Molecules cannot have more atoms than this number; implicit hydrogens do not count.
+        kekulize: bool, default True
+            Should molecules be kekulized; solves number of issues with defeaturization when used.
+        bond_labels: List[RDKitBond]
+            List containing types of bond used for generation of adjacency matrix
+        atom_labels: List[int]
+            List of atomic numbers used for generation of node features
+        """
+        self.max_atom_count = max_atom_count
+        self.kekulize = kekulize
+
+        # bond labels
+        if bond_labels is None:
+            self.bond_labels = [
+                Chem.rdchem.BondType.ZERO,
+                Chem.rdchem.BondType.SINGLE,
+                Chem.rdchem.BondType.DOUBLE,
+                Chem.rdchem.BondType.TRIPLE,
+                Chem.rdchem.BondType.AROMATIC,
+            ]
+        else:
+            self.bond_labels = bond_labels
+
+        # atom labels
+        if atom_labels is None:
+            self.atom_labels = [0, 6, 7, 8, 9]  # C,N,O,F
+        else:
+            self.atom_labels = atom_labels
+
+        # create bond encoders and decoders
+        self.bond_encoder = {l: i for i, l in enumerate(self.bond_labels)}
+        self.bond_decoder = {i: l for i, l in enumerate(self.bond_labels)}
+        # create atom encoders and decoders
+        self.atom_encoder = {l: i for i, l in enumerate(self.atom_labels)}
+        self.atom_decoder = {i: l for i, l in enumerate(self.atom_labels)}
+
+    def _featurize(self, mol: RDKitMol) -> GraphMatrix:
+        """Calculate adjacency matrix and nodes features for RDKitMol.
+
+        Parameters
+        ----------
+        mol: rdkit.Chem.rdchem.Mol
+          RDKit mol object.
+        Returns
+        -------
+        graph: GraphMatrix
+          A molecule graph with some features.
+        """
+        if self.kekulize:
+            Chem.Kekulize(mol)
+
+        A = np.zeros(shape=(self.max_atom_count, self.max_atom_count), dtype=np.float32)
+        bonds = mol.GetBonds()
+
+        begin, end = [b.GetBeginAtomIdx() for b in bonds], [
+            b.GetEndAtomIdx() for b in bonds
+        ]
+        bond_type = [self.bond_encoder[b.GetBondType()] for b in bonds]
+
+        A[begin, end] = bond_type
+        A[end, begin] = bond_type
+
+        degree = np.sum(A[: mol.GetNumAtoms(), : mol.GetNumAtoms()], axis=-1)
+        X = np.array(
+            [self.atom_encoder[atom.GetAtomicNum()] for atom in mol.GetAtoms()]
+            + [0] * (self.max_atom_count - mol.GetNumAtoms()),
+            dtype=np.int32,
+        )
+        graph = GraphMatrix(A, X)
+
+        return graph if (degree > 0).all() else None
+
+    def _defeaturize(
```

Comment:
I love this! We might even want to (in a separate PR) make a `defeaturize()` method in the `MolecularFeaturizer` superclass for a generalized notion of inversion

Reply:
We can do that, this is copy paste version of featurization method to make forward and backward conversion compatible with the base class. Alternative approach is to have single function that can be used by both.

Reply:
I will leave it as is for now and then move to new PR once everyone is happy with it.